### PR TITLE
Implement catalog endpoint

### PR DESCRIPTION
**Pull request by sahamanish99**
For Keivan bc his screen broke

### CODE REVIEW

File: src/common/utils.ts [hunk 1]
```diff
@@ -21,9 +24,102 @@ const generateHash = (input: string) =>
...
+      price: variation.item_variation_data.price_money.amount
+    };
+  });
+
+  return parsedVariations;
+};
+
+const parseModifierListIDs = (modifier_lists) => {
+  let parsedModifierLists = [];
+  if (modifier_lists != undefined) {
```
Comment (meganle) [inline src/common/utils.ts]: `!==`?

Comment (kevinchan159) [inline src/common/utils.ts]: should this be `!==`

File: src/common/utils.ts [hunk 2]
```diff
@@ -21,9 +24,102 @@ const generateHash = (input: string) =>
...
+  let data: ListCatalogResponse = await squareAPI.fetchCatalog();
+  let catalog = {
+    items: {},
+    modifier_lists: {},
+    taxes: {},
+  };
+
+  data['objects'].forEach(object => {
+    switch (object.type) {
+      case 'ITEM': {
```
Comment (kevinchan159) [inline src/common/utils.ts]: might be good to make these strings constants

File: src/common/utils.ts (hunk 1 above)
Comment (sahamanish99) [reply src/common/utils.ts]: can we just use `if (modifier_lists)` instead?

**Pull request merged**